helix-term/src/ui/editor.rs

```diff
@@ -1457,9 +1494,28 @@ impl Component for EditorView {
         if let Some(completion) = self.completion.as_mut() {
             completion.render(area, surface, cx);
         }
+
+        if let Some(explore) = self.explorer.as_mut() {
+            if explore.is_focus() {
+                let area = if use_bufferline {
```

Comment:
I think this code could be factored into local lambda so we don't repeat it all again.

Reply:
I think we can just have a second variable `let area_inner = ..` for this instead of a lambda. Actually it's a bit of an oversight from the bufferline implementation that this isn't already the case. IMO the bufferline should more or less transparently shrink the editors view area so we don't have to consider it everywhere. 

I this PR you shouldn't change existing code to account for that but introducing the `area_inner` binring here is nice future proofing IMO.

Reply:
I think we can just have a second variable `let area_inner = ..` for this instead of a lambda. In fact that is how it used to be before this PR so we should probably just create `let area_inner = ..` (using the logic here) and use that for the explorer and also in the existing `editor_area` calculation.

Reply:
Simply move this let binding to the start of the function and reuse it everywhere instead of redefining it each time (and give it a different name)